дз выполнено

loordbarringtn opened the pull request.

loordbarringtn committed:
дз выполнено

Severenit commented on main.js:
```diff
@@ -1,4 +1,8 @@
+const randomButton = document.querySelector('.button');
+
+const player1 = {
+    player: 1,
```
давай наверное переименуем это все таки в id или playerId, в марафоне все таки я не совсем корректно назвал эту переменную)

Severenit commented on main.js:
```diff
@@ -18,28 +23,61 @@ const name2 = {
+}
+
+function createPlayer(player) {
+    const playerElement = createElement('div', 'player'+player.player);
```
вижу ты уже используешь бектики, мультистроки, так почему бы здесь их не использвать?

Severenit commented on main.js:
```diff
@@ -18,28 +23,61 @@ const name2 = {
+    const progressbarEl = createElement('div', 'progressbar');
+    const lifeElement = createElement('div', 'life');
     lifeElement.style.width = `${player.hp}%`;
-    const nameElement = document.createElement('div');
+    const nameElement = createElement('div', 'name');
     nameElement.innerText = player.name;
-    nameElement.classList.add('name');
     progressbarEl.appendChild(lifeElement);
     progressbarEl.appendChild(nameElement)
     const characterElement = document.createElement('div');
     characterElement.classList.add('character');
-    const imageElement = document.createElement('img');
+    const imageElement= createElement('img');
```
давай все константы объявим в начале функции)

Severenit commented on main.js:
```diff
@@ -18,28 +23,61 @@ const name2 = {
-    document.querySelector('.arenas').appendChild(playerElement);
+function playerLoose(name) {
+    const  loseTitle = createElement('div', 'loseTitle');
+    loseTitle.innerText = name + ' lose';
```
тоже используй бектики раз уже знаком с ними
```suggestion
    loseTitle.innerText = `${name} lose`;
```

Severenit commented on main.js:
```diff
@@ -18,28 +23,61 @@ const name2 = {
+
+function playerWin(name) {
+    const  loseTitle = createElement('div', 'loseTitle');
+    loseTitle.innerText = name + ' wins!';
```
Тоже самое)

Severenit commented on main.js:
```diff
@@ -18,28 +23,61 @@ const name2 = {
+    player.hp -= Math.ceil(Math.random()*20);
+    playerLife.style.width = player.hp + '%';
+    if (player.hp <= 0) {
+        const winPlayer = player.player === 1 ? player2 : player1;
```
смотри у тебя функция changeHP помимо того что меняет hp она еще решает кто победил, и есть завязка на другого плеера

Сразу возникают вопросы: 
1. зачем тебе тогда плеер если у тебя и так тут есть доступ к player1 и 2
2. Если это функция которая меняет hp и решает кто выйграл должна ли она называться changeHP? может тогда она должна называться changeHpAndHowWin? ну это как предположение.
3. По заданию мы хотим чтобы эта функция была чистой, т.е. по факту, она должна принять какое-то значение и вернуть новое. 

Подумай об этом)

Severenit reviewed (commented):
Пока не могу поставить approve потому как есть критические замечания) относятся они к changeHP

loordbarringtn committed:
рефакторинг по замечаниям к pl:
переменная player->id
использование бектиков
функция changeHP теперь чистая
создана новая функция handleGameResult

Severenit commented on main.js:
```diff
@@ -18,28 +23,70 @@ const name2 = {
-createPlayer('player2', name2);
\ No newline at end of file
+
+function changeHP(player) {
+    player.hp -= Math.ceil(Math.random() * 20);
+    const playerLife = document.querySelector(`.player${player.id} .life`);
+    playerLife.style.width = player.hp + '%';
+}
```
но есть вероятность что у тебя может жизнь быть меньше 0 и тогда style не отработает) так что желательно эту проверку делать здесь)

Severenit commented on main.js:
```diff
@@ -18,28 +23,70 @@ const name2 = {
+function handleGameResult(player1, player2) {
+    let winner, loser;
+
+    if (player1.hp <= 0) {
+        winner = player2;
+        loser = player1;
+    } else if (player2.hp <= 0) {
+        winner = player1;
+        loser = player2;
+    }
+
+    if (loser) {
+        const looserLife = document.querySelector('.player' + loser.id + ' .life');
+        arenasElement.appendChild(playerWin(winner.name));
+        looserLife.style.width = 0 + '%';
+        randomButton.disabled = true;
+    }
```
Очень оригинальный способ про победителей) ладно оставь так если это работает и смотри просто дальше))

Severenit reviewed (approved):
Апрув поставлю) так как в целом сделал задание, но есть маленькие замечания, дальше в уроках найдешь на них ответ)
